src/json.rs

```diff
@@ -289,28 +297,29 @@ mod tests {
         assert_eq!(err, JsonError::NotJsonType);
     }
 
+    #[test]
+    #[should_panic]
+    fn test_json_type_u64_with_u128_value() {
+        let _ = parse_json(format!(r#"{{"foo": {} }}"#, u128::MAX).as_bytes());
```

Comment:
Shouldn't this parse as an `F64` if it doesn't fit in `U64`? Seems like panicking is too extreme.

Reply:
It's `rjson` library parser logic.
Also `u128::MAX > f64::MAX`
I extended with more explicit panuc message

Reply:
No, that is not correct. `f64::MAX` is much larger than `u128::MAX`. For example see code [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7ccfae9adc9068876d9a5bdf58fc4115).

I know it's coming from `rjson`. I am saying I think the `rjson` logic is wrong to panic here (in my opinion). But I don't know if there is a json spec or something which says how big numbers are allowed to be.

Reply:
According to https://www.json.org JSON numbers are always floats, but the spec doesn't give a a maximum size. A classic case of "implementation defined"...

Reply:
You are right that `f64` does not have the same precision as `u128`, but it can still represent bigger numbers.

The question is whether it is expected to panic if I pass in a number which is larger than a `u64` and has more precision than an `f64` can represent. I argue that most users would not expect a panic in this case and would instead expect a loss of precision. It is very normal in floating point arithmetic to have loss of precision (which of course is why we are trying to avoid floats in our use case).

Reply:
So seems we need put some changes to our `rjson` implementation.

Reply:
https://github.com/aurora-is-near/rjson

Reply:
Are you going to make those changes or should I? I suppose there is also the question of whether we should bother putting time into `rjson` at all if we will eventually move to using `serde` as @joshuajbouw works on integrating `std` into the contract.

Reply:
@birchmd @joshuajbouw - my suggestion to merge it **AS IS** - current implementation is ready for it.
But after some near period we will migrate to `serde`. And just do not spend additional time to `rjson` question.

Reply:
Currently, we spent a lot of time for `rjson`. I believe enough. The current solution is working. So the next step will be improvements with `serde`.